packages/backend/src/Application.ts

```diff
@@ -51,8 +54,14 @@ export class Application {
       createUpdateMonitorModule(config, logger, http, database, clock),
       createDiffHistoryModule(config, logger, database),
       createStatusModule(config, logger, database, clock),
-      createLivenessModule(config, logger, database, clock),
-      createFinalityModule(config, logger, database, clock),
+      livenessModule,
+      createFinalityModule(
+        config,
+        logger,
+        database,
+        clock,
+        livenessModule?.indexer as LivenessIndexer,
+      ),
```

Comment:
I'm not a big fan of this approach here. The whole point of creating those modules in an array is some separation of concerns. The problem here is that the finality module depends on the liveness module, which does not fit this style. What about creating a liveness and finality module as one?